### ISSUE TITLE
Which version of gast for TF 2.4?  (Autograph warning)

### ISSUE THREAD

**Issue by Mastiff37** [tensorflow/tensorflow#48889]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Centos7
- TensorFlow installed from (source or binary):  Conda install
- TensorFlow version: 2.4.1
- Python version: 3.7
- Installed using virtualenv? pip? conda?: Conda

**Describe the problem**

When training a model, receive "WARNING: Autograph could not transform..." ... "Cause: module 'gast' has no attribute 'Index'"

I have gast 0.4.0 installed.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

model.fit() with custom loss.

Sorry, can't paste long stuff in due to being on offline system.

My real question, do I need to back off to an older gats version?

**Comment by bhack** [comment]
Check https://github.com/tensorflow/tensorflow/issues/44146

**Comment by saikumarchalla** [comment]
@Mastiff37   please use gast 0.3.3 for Tensorflow version 2.4. If you prefer to work with gast 0.4 use latest Tensorflow/Nightly version. Thanks!

**Comment by Mastiff37** [comment]
When I attempt to install gast 0.3.3 with conda, it wants to downgrade tensorflow-gpu to 2.2.  Is this expected?

**Comment by bhack** [comment]
We don't directly support conda install here.

**Comment by Mastiff37** [comment]
Ok thanks.  This would seem to indicate that TF 2.4 is somehow telling conda it needs a newer version of gast (than 0.3.3).  My understanding is limited, but don't the TF developers provide the dependencies to conda somehow?

**Comment by bhack** [comment]
If you need Conda it is supported by third-party. 
You could try to ask or open a ticket at:
https://docs.anaconda.com/anaconda/user-guide/tasks/tensorflow/
https://github.com/AnacondaRecipes/tensorflow_recipes

**Comment by saikumarchalla** [comment]
@Mastiff37  Please go ahead and close the issue if you don't have any further questions. Thanks!

**Comment by samarth-robo** [comment]
> When I attempt to install gast 0.3.3 with conda, it wants to downgrade tensorflow-gpu to 2.2. Is this expected?

@Mastiff37 I ran `pip install gast==0.3.3` inside the Conda environment. It complained about dependencies but downgraded anyway. It seems to solve the issue.

**Comment by Mastiff37** [comment]
> > When I attempt to install gast 0.3.3 with conda, it wants to downgrade tensorflow-gpu to 2.2. Is this expected?
> 
> @Mastiff37 I ran `pip install gast==0.3.3` inside the Conda environment. It complained about dependencies but downgraded anyway. It seems to solve the issue.

Yes.  I'm learning that sometimes conda is just too restrictive.  pip does what it's asked.  My MO now is to use conda until I can't, then just start pipping.
